packages/kit/src/exports/node/index.js

```diff
@@ -28,10 +27,10 @@ function get_raw_body(req, body_size_limit) {
 		if (!length) {
 			length = body_size_limit;
 		} else if (length > body_size_limit) {
-			error(
-				413,
-				`Received content-length of ${length}, but only accept up to ${body_size_limit} bytes.`
-			);
+			throw {
+				status: 413,
+				message: `Received content-length of ${length}, but only accept up to ${body_size_limit} bytes.`
+			};
```

Comment:
Why doesn't this use `SvelteKitError`?

```suggestion
			throw new SvelteKitError(413, 'Payload Too Large', `Received content-length of ${length}, but only accept up to ${body_size_limit} bytes.`);
```

Reply:
I don't think an error at this stage will be handled by `handelError`, so I kept this clear of it. But we could use it.

Reply:
Not at _this_ stage, no. But the thing that reads the body will either catch the error (in which case it is then responsible for setting a status code) or, more likely, the error will go uncaught. At that point, it's useful if the error was a `SvelteKitError` with a correct status code. If it's a POJO then `handleError` will be called with a 500 Internal Error.

Reply:
More correctly: the above applied to the second occurrence, once the stream has already started being read.

In _this_ case, we're still creating the `Request` object. Right now, every usage of `getRequest` has to be wrapped in a `try-catch`, and the `catch` clause always looks like this:

```js
res.statusCode = err.status || 400;
res.end('Invalid request body');
```

This makes no sense — there's only one possible error (the 413), and the diagnostic message gets swallowed.

I think the best solution is to throw the 'content length exceeds limit' error immediately when the stream starts. That way, `getRequest` callers don't need to `try-catch`, and we get both user-friendly and developer-friendly error messages.